components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/authz/OAuth2AuthzEndpoint.java

```diff
@@ -2586,6 +2593,49 @@ private String appendAuthenticatedIDPs(SessionDataCacheEntry sessionDataCacheEnt
         return redirectURL;
     }
 
+    /**
+     * Store opbscookie in session context.
+     *
+     * @param sessionDataCacheEntry SessionDataCacheEntry.
+     * @param opbscookie            opbscookie value.
+     */
+    private void storeOpbsInSessionContext(SessionDataCacheEntry sessionDataCacheEntry, String opbscookie) {
+
+        String sessionContextIdentifier = getSessionContextIdentifier(sessionDataCacheEntry);
+        if (StringUtils.isNotBlank(sessionContextIdentifier)) {
+            SessionContext sessionContext = FrameworkUtils.getSessionContextFromCache(sessionContextIdentifier);
+            if (sessionContext != null) {
```

Comment:
shall we merge the nested if condition with this?

Reply:
there are two else part here